hack/ensure_job_cluster.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+"""This script ensures the cluster of prowjobs is defined as expected.
+e.g., python ./hack/ensure_job_cluster.py
+"""
+
+import argparse
+import logging
+import os
+import sys
+import yaml
+
+
+def get_desired_cluster(file_path, job):
+    """get the desired cluster for a given job defined in file_path."""
+    if job.get("agent", "") != "kubernetes":
+        return ""
+    if job["name"] in JOB_MAP:
+        return JOB_MAP[job["name"]]
+    if job["name"].endswith("-build01") or migrated(file_path):
+        return BUILD01_CLUSTER
+    return ""
+
+
+def handle(file_path, jobs):
```

Comment:
Right now the decision on whether you are updating or checking happens in a couple places. Try to simplify this. 

`handle()` is a bad name for a function, since it just tells you that you're "doing everything" -- this is part of the symptom. If we instead were to write `identify_jobs_to_update()` you could have a simple function that does one thing and one thing only, then the caller of that could determine if they should emit an error or update the file.

Reply:
Not sure if i did it right, leaving this comment unresolved.